packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,197 @@
+            const jsonData = decoratorDict[namespace];
+            const patternToDetermineVocab = /^Term_/i;
+            jsonData.forEach(obj =>{
+                const decos = JSON.parse(obj.dcs);
+                const target = {
+                    '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.CommandTarget`,
+                    'namespace':namespace
+                };
+                if (obj.declaration && obj.declaration!==''){
+                    target.declaration = obj.declaration;
+                }
+                if (obj.property && obj.property!==''){
+                    target.property = obj.property;
+                }
+                if (obj.mapElement && obj.mapElement!==''){
+                    target.mapElement = obj.mapElement;
+                }
+                decos.forEach(dcs =>{
+                    if (dcs.name !== 'Term' && !patternToDetermineVocab.test(dcs.name)){
+                        const decotatorObj = {
+                            '$class': 'concerto.metamodel@1.0.0.Decorator',
+                            'name': dcs.name,
+                        };
+                        if (dcs.arguments){
+                            const args = dcs.arguments.map((arg)=>{
+                                return {
+                                    '$class':arg.$class,
+                                    'value':arg.value
+                                };
+                            });
+                            decotatorObj.arguments = args;
+                        }
+                        let dcsObject = {
+                            '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.Command`,
+                            'type': 'UPSERT',
+                            'target': target,
+                            'decorator': decotatorObj,
+                        };
+                        dcsObjects.push(dcsObject);
+                    }
+                });
+            });
+            const dcmsForNamespace = {
+                '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.DecoratorCommandSet`,
+                'name': nameOfDcs,
+                'version': versionOfDcs,
+                'commands': dcsObjects
+            };
+            data.push(dcmsForNamespace);
+        });
+        return data;
+    }
+    /**
+     * Parses the dict data into an array of decorator jsons
+     * @param {Object} decoratorDict the input dict
+     * @param {String} locale locale for target vocab
+     * @returns {Array<Object>} the parsed decorator command set array
+     * @private
+     */
+    static parseVocabularies(decoratorDict,locale){
+        const data = [];
+        const patternToDetermineVocab = /^Term_/i;
+        Object.keys(decoratorDict).forEach((namespace)=>{
+            if (JSON.stringify(decoratorDict[namespace]).includes('Term') ||
+            JSON.stringify(decoratorDict[namespace]).includes('Term_')){
```

Comment:
Why call `JSON.stringify` two separate times? I'm pretty sure it's a relatively expensive operation. You can form the string before the if statement.

Reply:
Also if this is needed because `Term` is deep in the data structure, you can consider flattening the structure and looping over those decorators?

Reply:
Figured a way to do without the stringify check